tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java

```diff
@@ -7346,4 +7354,24 @@ public static String utf8truncate(String input, int length) {
 	return result.toString();
     }
 
+	/**
+	 * Method to add Resources folder into Lessons tool
+	 */
+	public String folderPickerSubmit(){
+		//Check if user has submitted page without selecting folder?
+		String defaultPath = "/group/" + getCurrentSiteId();
+		if(folderPath == null ||  folderPath.equals(defaultPath)){
+			return "failure";
+		}
+		String dataDirectory = defaultPath + "/" + folderPath;
+		contents = "<div data-copyright=\"true\" class=\"no-highlight\" data-description=\"true\" data-directory=" +dataDirectory+ " data-files=\"true\" data-folder-listing=\"true\"></div>";
+		Placement placement = toolManager.getCurrentPlacement();
+		//setting filter false to disable antisamy for folder listing
+		placement.getPlacementConfig().setProperty("filterHtml","false");
```

Comment:
Ok, so although it's in the same file changing a thread local variable before a method call is very un-obvious to anyone changing `submit()`. If you passed a boolean in the parameters of `bypassFiltering` it's much clearer that there is a way to override the HTML filtering.

To expand on what I meant in my first comment is that there is nothing in the contract of the `ToolManager` that if you make changes to the placement you receive from the `getCurrentPlacement()` call you should expect to see those changes when you get the current placement in the future. It may work for now, but if someone changes how the API is implemented it's the sort of thing that may break in the future.

All that said, if you don't want any filtering it might be much clearer to just create the SimplePageItem yourself and call the `update(SimplePageItem)` method rather than complicating the `submit()` method with more shortcuts.